packages/components/src/components/permissions/PermissionsContextProvider.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2018 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
+ * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
+ */
+import * as React from 'react'
+import { List, Map, fromJS } from 'immutable'
+import { Security } from '@labkey/api'
+import { PermissionsProviderProps, Principal, SecurityRole } from "./models";
+import { ISelectRowsResult, selectRows } from "../../query/api";
+import { LoadingPage } from "../../components/base/LoadingPage";
+
+const Context = React.createContext<PermissionsProviderProps>(undefined);
+const PermissionsContextProvider = Context.Provider;
+export const PermissionsContextConsumer = Context.Consumer;
+
+type Props = PermissionsProviderProps;
+
+interface State extends PermissionsProviderProps {}
+
+export const PermissionsPageContextProvider = <P extends PermissionsProviderProps>(Component: React.ComponentType<P>) => {
+
```

Comment:
Since the LABKEY.user object doesn't get refreshed unless the app is reloaded, it seems possible that the current user's permissions will be reflected differently by this provider than by that object.  Should we issue a refresh at some point after updating permissions?

Reply:
This provider does get reloaded with each route navigation into one of these related pages in the app. However, related to what you are saying, if I as an App Admin change your permissions while you are in the app, your client side code won't know about that but any API calls you make will. Not sure there is much we can do about that though, do you?

Reply:
The provider gets reloaded, but does the LABKEY object get reloaded without an app reload?

For your question, I think our recourse there is to make sure we behave gracefully when the permission errors come back from the server.  Currently, unfortunately, we definitely do not.  But that's for a different story.